specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2020-03-01/machineLearningServices.json

```diff
@@ -1228,6 +1228,283 @@
           }
         }
       }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/Microsoft.MachineLearningServices/workspaces/{workspaceName}/linkedWorkspaces": {
+      "get": {
+        "tags": [
+          "LinkedWorkspaces"
+        ],
+        "x-ms-examples": {
+          "ListLinkedWorkspaces": {
+            "$ref": "./examples/listLinkedWorkspaces.json"
+          }
+        },
+        "operationId": "LinkedWorkspaces_List",
+        "description": "List linked workspaces under a workspace per some query criteria.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/WorkspaceNameParameter"
+          },
+          {
+            "name": "link-subscriptionId",
+            "in": "query",
+            "type": "string",
+            "description": "The subscriptionid of the linked workspace."
+          },
+          {
+            "name": "link-resource-group",
+            "in": "query",
+            "type": "string",
+            "description": "The resource group name of the linked workspace."
+          },
+          {
+            "name": "link-resource-name",
+            "in": "query",
+            "type": "string",
+            "description": "The resource name of the linked workspace."
+          },
+          {
+            "name": "link-resourceId",
+            "in": "query",
+            "type": "string",
+            "description": "The resourceId of the linked workspace."
+          },
+          {
+            "name": "link-type",
+            "in": "query",
+            "type": "string",
+            "description": "The type of the linked workspace.",
+            "enum": [
+              "Synapse"
+            ],
+            "x-ms-enum": {
+              "name": "LinkType",
+              "modelAsString": true
+            }
+          },
+          {
+            "name": "uai-resourceId",
+            "in": "query",
+            "type": "string",
+            "description": "The resourceId of the user assigned identity for the linked workspace."
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successfully retrieved the linked workspaces.",
+            "schema": {
+              "type": "array",
+              "description": "array of linked workspaces.",
+              "items": {
+                "$ref": "#/definitions/LinkedWorkspace"
+              }
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed",
+            "schema": {
+              "$ref": "#/definitions/MachineLearningServiceError"
+            }
+          }
+        }
+      },
+      "post": {
```

Comment:
>post [](start = 7, length = 4)

POST is not allowed on collections as per the RPC. Also, resource updates are to be done via PUT and optionally PATCH verbs.

Reply:
The item in the collection is not a resource, so are we able to use PUT/PATCH/DELETE for a non-resource item?

Reply:
A "resource" doesn't have to be anything tangible. A resource is anything with a lifecycle. That is, anything that you can create, read, list, delete, etc.

The GET, PUT, PATCH, DELETE verbs in ARM are reserved for resources. POST is reserved for actions. Looking at the PR, my recommendation would be to model linkedWorkspaces as a resource.

Reply:
We're very likely to make LinkedWorkspace a resource during our public preview stage. For private preview, we'd like keep it a non-resource item to roll out our beta faster. Your understanding and support is appreciated.
I knew the PUT/PATCH/DELETE is reserved for resource, that's why I use POST here.

Reply:
Add one more justification for not making LinkedWorkspace resource for now.
We may change the model of how we do this linking integration (e.g. we may do it in the way of ADF linked service) after we get feedback from customer, which is the main reason that we don't want to invest on creating sub-resource for now.

Reply:
When you have a path like `/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/Microsoft.MachineLearningServices/workspaces/{workspaceName}/linkedWorkspaces`, the user expectation is that GET `/linkedWorkspaces` will return a collection of nested resources. This is because this pattern is repeated in all APIs. There is no way for the user to know that this is "not a resource" on this particular API

If it was a POST, it would be something like POST `/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/Microsoft.MachineLearningServices/workspaces/{workspaceName}/listLinkedWorkspaces`, then it would be clear that this is an action that lists some things that aren't necessarily resources. This is implied by POST and the imperative name of the action. (Regardless doing a list* action via POST is not discouraged because it causes user confusion with RBAC, prevents enforcement by policy, prevents deployments via ARM templates, etc.)

---
In reply to: [403138744](https://github.com/Azure/azure-rest-api-specs/pull/8918#discussion_r403138744) [](ancestors = 403138744)

Reply:
Got you. So I suppose if I change the CUD like below it should be ok?
Create: POST .../{workspaceName}/linkWorkspace
Update: POST .../{workspaceName}/updateLinkedWorkspace/{linkName}
Delete: POST .../{workspaceName}/delinkWorkspace/{linkName}
And as of get and list, per your reply, we can use:
Get: POST .../{workspaceName}/GetLinkedWorkspace/{linkName}
List: POST .../{workspaceName}/listLinkedWorkspaces

And when we change the LinkedWorkspace to be a resource later, we will follow all the rest standard of ARM. Are you good with that?